static/elements/chromedash-callout.js

```diff
@@ -0,0 +1,136 @@
+import {LitElement, css, html} from 'lit-element';
+
+class ChromedashCallout extends LitElement {
+  static get properties() {
+    return {
+      signedin: {type: Boolean},
+      targetid: {type: String},
+      // TODO(jrobbins): Support sides other than "south".
+      side: {type: String}, // "north", "south", "east", or "west"
+      hidden: {type: Boolean},
+      top: {type: Number},
+      left: {type: Number},
+      cue: {type: String}, // String to send to the server when dismissed.
+      dismissedcues: {attribute: false},
+    };
+  }
+
+  constructor() {
+    super();
+    this.side = 'south';
+    this.hidden = true;
+    this.top = 0;
+    this.left = 0;
+    this.signedin = false;
+    this.dismissedcues = [];
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    if (this.dismissedcues.includes(this.cue)) {
+      // Don't show the cue because the user has already dismissed it.
+      return;
+    }
+    let targetEl = this.parentNode.querySelector('#' + this.targetid);
+    if (targetEl) {
+      this.attach(targetEl);
+    } else {
+      console.log('Callout target not found: #' + this.targetid);
+    }
+  }
+
+  static get styles() {
+    return css`
+      :host {
+        /* TODO(jrobins): Put this into a global theme.css file. */
+        --blue-900: #174EA6;
+
+        --callout-bg-color: var(--blue-900);
+        --callout-text-color: white;
+      }
+      #bubble {
+        position: absolute;
+        display: inline-block;
+        background: var(--callout-bg-color);
+        border-radius: 8px;
+        padding: 8px;
+        max-width: 24em;
+        color: var(--callout-text-color);
+        font-weight: bold;
+      }
+      #bubble[hidden] {
+        display: none;
+      }
+      #bubble:after {
+        content: '';
+        position: absolute;
+        border: 20px solid transparent;
+      }
+      /* Bubble will be located on the south side of target element. */
+      #bubble.south:after {
+        top: 0;
+        left: 40px;
+        width: 0;
+        height: 0;
+        border-bottom-color: var(--callout-bg-color);
+        border-top: 0;
+        margin-left: -20px;
+        margin-top: -20px;
+     }
+     #closebox {
+       float: right;
+       margin-left: 8px;
+     }
+    `;
+  }
+
+  contentOffsetTop(el) {
```

Comment:
It looks like getting the Y position of an element on the page?

`el.getBoundingClientRect().top + window.scrollY` might be better.

Reply:
That calculation was the part of the CL that I struggled with the most.  I considered using getBoundingClientRect, but then I wasn't sure about the coordinate system reference point.  The callout itself is position:absolute, which means that its position is relative to its lowest positioned ancestor.  Whereas, getBoundingClientRect and window.scrollY would be relative to the document.  So, wouldn't I need to dynamically create the chromedash-callout element as a child of the document rather than somewhere deep in the DOM?

For now, I am sticking with what I have because it works.  I added a TODO to come back to try a simpler design.

Reply:
getBoundingClientRect is relative to the viewport, so it shouldn't be affected by position property on itself, or its parent/ancestors.

SGTM to keep it as is, and revisit it in the future.